app/models/katello/activation_key.rb

```diff
@@ -76,27 +79,15 @@ def available_releases
       end
     end
 
-    # For efficiency, sometimes the candlepin pool objects have already been fetched so allow
-    # them to be passed in directly. By default, a call to candlepin will be made
-    def subscriptions(cp_pools = nil)
-      cp_pools ||= self.get_key_pools
-
-      pools = cp_pools.collect { |cp_pool| Pool.find_pool(cp_pool['id'], cp_pool) }
-
-      subscriptions = pools.collect do |pool|
-        product = Product.where(:cp_id => pool.product_id).first
-        next if product.nil?
-        pool.provider_id = product.provider_id
-        pool
-      end
-      subscriptions.compact
+    def subscriptions
+      self.pools
     end
 
     def available_subscriptions
-      all_pools = self.get_pools
-      key_pool_ids = self.get_key_pools.collect { |pool| pool[:id] }
-      pools = all_pools.reject { |pool| key_pool_ids.include? pool[:id] }
-      self.subscriptions(pools)
+      all_pools= self.get_pools.map { |pool| pool["id"] }
```

Comment:
What I have now is getting the available subscriptions JSON from candlepin, I was torn between doing that and returning from the AR database like you are suggesting, I thought returning them from candlepin left less room for error? Do you feel its better to return from AR?